src/sync/event-status-withdrawing-changed.js

```diff
@@ -0,0 +1,112 @@
+const { each, isNull, find } = require('lodash');
+const web3 = require('../web3');
+const { TX_STATUS } = require('../constants');
+const EventSig = require('../config/event-sig');
+const { toLowerCase } = require('../utils');
+const { getTransactionReceipt } = require('../utils/web3-utils');
+const logger = require('../utils/logger');
+const DBHelper = require('../db/db-helper');
+const parseEvent = require('./parsers/multiple-results-event');
+const MultipleResultsEventApi = require('../api/multiple-results-event');
+const EventLeaderboard = require('../models/event-leaderboard');
+const GlobalLeaderboard = require('../models/global-leaderboard');
+
+const getInvestments = async (txs) => {
+  const accumulated = txs.reduce((acc, cur) => {
+    const amount = web3.utils.toBN(cur.amount);
+    if (!cur.betterAddress) cur.betterAddress = cur.centralizedOracleAddress;
+    if (Object.keys(acc).includes(cur.betterAddress)) {
+      acc[cur.betterAddress] = web3.utils.toBN(acc[cur.betterAddress]).add(amount).toString(10);
+    } else {
+      acc[cur.betterAddress] = amount.toString(10);
+    }
+    return acc;
+  }, {});
+
+  return accumulated;
+};
+
+const updateLeaderboardWithdrawing = async (syncPromises, events, limit) => {
+  each(events, async (event) => {
+    syncPromises.push(limit(async () => {
```

Comment:
need to pass `event` to the `limit(...func, event)`. see https://github.com/bodhiproject/bodhi-server/pull/270/files#diff-ba9f3a5bc0319fb93654cb9d99155063R24

Reply:
ok will do.

````Support for passing arguments on to the fn is provided in order to be able to avoid creating unnecessary closures. You probably don't need this optimization unless you're pushing a lot of functions.```

But I don't quite understand what will happen if no such optimization.

Reply:
It seems to be primarily for reference issues. In this situation, the promises are holding a ref to the `event` outside of promise itself. So in regards to garbage collection, the promise it forcing it to hold an a reference outside of the promise.

If you passed the event to the limit func, it wouldn't need that ref outside of it since its being passed in.

Reply:
Also, I verified that you dont need to pass funcs to the limit func since any funcs called within the limit func will inherently be limited as well.